app/src/main/AndroidManifest.xml

```diff
@@ -18,6 +18,7 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".SpeedTestKotlin" android:exported="true"/>
```

Comment:
- You create new Activity but cannot select both activity in runtime. I think you should be creating ListActivity to select both activity. I appreciated committing this Activity.